src/eval-frame/tools/send-file-request-to-editor.js

```diff
@@ -0,0 +1,39 @@
+import generateRandomId from "../../shared/utils/generate-random-id";
+import messagePasserEval from "../../shared/utils/redux-to-port-message-passer";
+
+export const fileRequests = {};
+
+export default function sendFileRequestToEditor(
```

Comment:
actually, looking into this more, i don't believe `fileRequestId` used anywhere except for in testing code, and for the testing case i think you could just mock `generateRandomId` to return a fixed string. let's remove that if it's only used for testing

Reply:
Not true – `fileRequestId` is needed to actually resolve the file request. File name alone is not quite enough, because a user can send off two file requests for the same file at the same time. This minimally allows you to handle these issues gracefully, since the request itself is the unit of interest, not the file name.

This said, 100% right about moving to async / await here. I neglected to do that on the eval frame side - good catch!

Reply:
ya i know it's used to resolve the file request, but it looks like in that the value of `fileRequestId` is recieved by `onParentContextFileRequestSuccess` and `onParentContextFileRequestError` in _messages_ that come back from the editor, _not_ as the direct return value of `sendFileRequestToEditor`. i don't think that the `fileRequestID` value that is directly returned from `sendFileRequestToEditor` is used anywhere (and as a conceptual matter, it does not seem like it would need to be used anywhere, since the point of the ID is to set up a resolver that waits for the ID and send the ID over to the editor -- only the resolver needs to be able to see see the ID, nothing else in the eval-frame should need to be aware of it until the message comes back, right?)

Reply:
oh I see your point. I can actually generate the ID as a default arg like we do pretty much everywhere else and that shouldn't affect the testability (seems pretty obvious now :) ). That all said, with the sum of simplifications, we don't really need async here since this function should just return a simple promise.

Reply:
i'd still like you to address this one if you don't mind. all async functions will return a promise automatically (wrap a return value in a promise) but for code readabilty i'd like us to never explicitly return a `Promise` in the code when we could just as easily write a function that returns a value.

i think the snippet below is clearer about what the intent of this function:
- we post a "FILE_REQUEST" message to the editor
- we await the result of the request
- we return the request result to the caller

```
export default async function sendFileRequestToEditor(
  fileName,
  requestType,
  options,
  fileRequestID = generateRandomId()
) {
  messagePasserEval.postMessage("FILE_REQUEST", {
    fileName,
    requestType,
    fileRequestID,
    options
  });

  const requestResult = await new Promise((resolve, reject) => {
      fileRequests[fileRequestID] = { resolve, reject };
  })

  return requestResult
}
```

As is, this function looks very mysterious to my eyes -- we're returning a promise that does some kind of funky thing that adds an object to the `fileRequests` object, but i need to go look at the call sites to understand the fate of that promise and what it is for.